Formula/jfrog-cli.rb

```diff
@@ -17,14 +18,17 @@ class JfrogCli < Formula
   depends_on "go" => :build
 
   def install
-    system "go", "build", "-ldflags", "-s -w -extldflags '-static'", "-trimpath", "-o", bin/"jfrog"
+    system "go", "build", *std_go_args(ldflags: "-s -w -extldflags '-static'", output: bin/"jf")
     prefix.install_metafiles
+    bin.install_symlink "jf" => "jfrog"
+
     system "go", "generate", "./completion/shells/..."
     bash_completion.install "completion/shells/bash/jfrog"
-    zsh_completion.install "completion/shells/zsh/jfrog" => "_jfrog"
+    zsh_completion.install "completion/shells/zsh/jfrog"
```

Comment:
Does this work? Does it not need to be `_jf`, or something?

Reply:
The Zsh completion executable name doesn't matter if I source it manually:
<img width="818" alt="image" src="https://user-images.githubusercontent.com/11367982/152652990-c4bef899-ac1f-4823-818d-e63f66bda8aa.png">


However, I can't make it work on my machine after installing with Homebrew. Do you have any idea why?
I followed the instructions here - https://docs.brew.sh/Shell-Completion#configuring-completions-in-zsh. Opening a new terminal doesn't help too.

Reply:
Did you try clearing your zsh completions cache?

Reply:
Yes. As suggested in the link, I ran `rm -f ~/.zcompdump; compinit`.
I'm using `oh-my-zsh` and I deleted its cache too.
I'm probably missing something, because the Zsh completion file does exist under `/usr/local/share/zsh/site-functions/jfrog` after installing with Homebrew. 🤔

Reply:
That's probably why it needs to be called `_jf`. Scripts in `site-functions` are not sourced automatically but instead made available to `FPATH`. That allows lazy loading of scripts to improve shell startup speeds.

Reply:
@carlocab I changed it to `_jf`. 
I also found the root cause of the issue: In order for Zsh completion script to be autoloadable by compinit, it should contain the `#compdef name` header. I created this PR to fix it in our side: https://github.com/jfrog/jfrog-cli/pull/1427.